src/fsharp/FSharp.Core/array.fs

```diff
@@ -478,6 +478,17 @@ namespace Microsoft.FSharp.Collections
         [<CompiledName("Where")>]
         let where f (array: _[]) = filter f array
 
+        [<CompiledName("Except")>]
+        let except (array1:'T[]) (array2:'T[]) =
+            checkNonNull "array1" array1
+            checkNonNull "array2" array2
+
+            if array1.Length = 0 then
```

Comment:
Yes the second argument could be seq (same for list) and I don't think it would be any less (in)efficient.  We made a similar decision for windowed, but not elsewhere to my knowledge